zndraw/zndraw_worker.py

```diff
@@ -367,42 +366,5 @@ def upload(self, target: str):
         """Emit all frames to the target (webclient)."""
         if not self.emit:
             raise ValueError("Emit is disabled")
-        config = GlobalConfig.load()
-        frame_list = []
-        for idx in range(len(self)):
-            frame_list.append(self[idx])
-            if len(frame_list) == config.read_batch_size:
-                msg = CeleryTaskData(
-                    target=target,
-                    event="room:set",
-                    data=RoomSetData(
-                        frames={
-                            idx
-                            - jdx: ZnFrame.from_atoms(atoms).to_dict(
-                                built_in_types=False
-                            )
-                            for jdx, atoms in enumerate(reversed(frame_list))
-                        },
-                        step=idx if idx < self.step else self.step,
-                    ).to_dict(),
-                )
-
-                self.socket.emit("celery:task:emit", msg.to_dict())
-                frame_list = []
-
-        msg = CeleryTaskData(
-            target=target,
-            event="room:set",
-            data=RoomSetData(
-                frames={
-                    idx - jdx: ZnFrame.from_atoms(atoms).to_dict(built_in_types=False)
-                    for jdx, atoms in enumerate(reversed(frame_list))
-                },
-                selection=self.selection,
-                points=self.points.tolist(),
-                bookmarks=self.bookmarks,
-                step=self.step,
-            ).to_dict(),
-        )
 
-        self.socket.emit("celery:task:emit", msg.to_dict())
+        self.extend(list(self))
```

Comment:
What would be the use case for doing `worker.upload` now? It seems like it duplicates the data, doesn't it?

Reply:
removed it